src/atomsWithQueryAsync.ts

```diff
@@ -0,0 +1,56 @@
+import type { QueryKey, QueryObserverOptions } from '@tanstack/query-core'
+import type { ExtractAtomValue, Getter, WritableAtom } from 'jotai'
```

Comment:
With https://github.com/pmndrs/jotai/pull/1741, we should be able to use it.
```suggestion
import type { ExtractAtomArgs, ExtractAtomValue, Getter, WritableAtom } from 'jotai'
```